### ISSUE TITLE
Gadgetzan

### ISSUE THREAD

**Issue by Phatcat** [classicdb/database#860]
The guards of Gadgetzan do not care about you attacking flight masters (at least the horde one) - not even the taxis he summons are willing to help him unless you enter their actual aggro range.
This may be a core issue.

Goblin engineer in Gadgetzan will teach you goblin recipies even if you're not specialized in goblin engineering.
He is willing to teach you even if you don't know engineering at all.

Update: 

~~Tombstone in Gadgetzan graveyard geets you with the text "Greetings <name>" - This doesn't seem intended.~~ solved in https://github.com/cmangos/classic-db/commit/b7410628df04b3d6d534bdd08e4d7e2cdd56af72

~~Rumsen Fizzlebrack (just right out of Gadgetzan) has a repeatable quest available to everyone - It's funny I never noticed this quest during vanilla, I did love the hell out of Tanaris and Zul Farrak back then. Then I read the quest (which is by the way "War at Sea") and it had this to say:~~

~~> You seek to befriend the goblins of Tanaris once again, name? (...)~~

~~Yeah, this quest (or any like it) should only be available to anyone at war with the faction in question (This is why he's placed outside the city, away from any guards - I had not even interacted with gadgetzan before this quest was available to me)~~ solved in https://github.com/cmangos/classic-db/commit/03d69306d39ee2606f7c74f801a37293d49d8162

**Comment by lduguid** [comment]
Concerning, _'Tombstone in Gadgetzan graveyard geets you with the text "Greetings " - This doesn't seem intended._', I do not know what the intended greeting should be when clicking on quest items when you are not on a relevant quest and/or appropriate part of quest chain but as an FYI, one of the tombstone's at the Gadgetzan graveyard is an item in the long quest chain involving Linken (http://db.vanillagaming.org/?npc=8737), this particular quest being 'Meet at the grave' (http://db.vanillagaming.org/?quest=3912).

**Comment by Phatcat** [comment]
Oh, I know it's an object relating to a quest, but it should be turned 'un-interactive' when not on the quest. Pretty sure you're able to set prerequesties for objects, no?

I'm new here on the db team so I am not really sure how to handle this or I would have taken care of that one myself.

**Comment by Tobschinski** [comment]
https://github.com/classicdb/database/issues/142

**Comment by Phatcat** [comment]
Yeah, I thought I'd seen it somewhere before... so it's a general issue we have that ~~we are missing pre requirements for a bunch of objects in the world?~~ I should learn to read better

**Comment by cala** [comment]
> The guards of Gadgetzan do not care about you attacking flight masters (at least the horde one) - not even the taxis he summons are willing to help him unless you enter their actual aggro range.
> This may be a core issue and if it is then I'm going to take a crack at it

Depending on how they are summoned (I don't have the NPC entry at hand, so I can't check):
- by spell: they should be considered as minion or guardians and inherit their master aggro list on spawn. In this case your soon-to-be-tested last PR about pets should take care of it IIRC the commits that are in
- by script: in this case, summoned do not inherit their master aggro list (and this is correct, there are encounters where the boss summons adds though they do not aggro on spawn). We could do a specific case for this NPC (and similar ones), but I think we may be summoning them the wrong way.

**Comment by Phatcat** [comment]
I'm just gonna take a wee visit back to Gadgetzan on that branch to see if those aggro commits are helpful in this case.

**Comment by Phatcat** [comment]
Okay, either the commits do not function properly or his taxis are scripted because they simply do not care.

**Comment by Phatcat** [comment]
Or.... It may be because he enters combat before doing his summons? in which case my branch does not help at all since the relevant commits effect takes place at entering combat only.

**Comment by cala** [comment]
> in which case my branch does not help at all since the relevant commits effect takes place at entering combat only

This should also work when master is already in combat.
Could you provide me the name or entry of the NPC, please? I will look at his script if any.

**Comment by Phatcat** [comment]
Of course, cala.

Bulkrek Ragefist - ID 7824

**Comment by cala** [comment]
Thanks. As I thought the script is (partly) faulty. In ACID, the NPC is directly summoning the NPC 9297 (Enraged Wyvern) instead of using what I assume to be the correct spell: 14307 - Summoned Enraged Wyvern.
This should provide better results (but probably not perfect):

``` SQL
DELETE FROM  `creature_ai_scripts` WHERE `id` = 782401;
INSERT INTO `creature_ai_scripts` VALUES (782401, 7824, 4, 0, 100, 0, 0, 0, 0, 0, 11, 14307, 0, 0, 1, -548, 0, 0, 0, 0, 0, 0, 'Bulkrek Ragefist - Summon Enraged Wyverns and Yell on Aggro');
```

**Comment by Phatcat** [comment]
Those ACID scripts that are flawed should of course be corrected to our best ability (replace any manual summon we can find with dedicated spells)

To fix the rest I can add in the 'assist summoner on summon' functionality I added to the other spell effects. It's because that functionality is not added for TempSummons being summoned manually (so far it has only been added to spelleffects.cpp)

**Comment by cala** [comment]
I'll update the ACID scripts soon, the spells exist in Classic DBCs and these summons were removed after, so we should not need a specific core update for this, fortunately. :)

**Comment by Phatcat** [comment]
The guards of gadgetzan still couldn't care less about you attacking flight masters (I'm guessing this is possibly core issue, maybe to do with generic GuardAI functions), the 2 bruisers next to the horde guy is purely for show on cmangos. His taxis are however extremely aggressive now.

This time I made sure I wasn't on gm mode and I was still able to learn goblin engineering recipies even though the char wasn't a goblin engineer <- this is a core thing isn't it?

the "War at Sea" quest, along with the ratchet, everlook and booty bay counterparts must not be available to anyone who is neutral (or higher) or unknown (hasn't interacted with them or gotten +rep for them yet) - these npcs are meant to be used to be bring your rep back up if you've been at war with them, the quest text for War at Sea even reads at the bottom

> "(...) and we'll be on our way to pardoning your tresspasses"

this would make no sense in the context they are available on cmangos (also they can be aused for 'cheap' rep grinding as it currently is, and it is a bit immersion breaking.

**Comment by Phatcat** [comment]
> The guards of gadgetzan still couldn't care less about you attacking flight masters (I'm guessing this is possibly core issue, maybe to do with generic GuardAI functions), the 2 bruisers next to the horde guy is purely for show on cmangos. His taxis are however extremely aggressive now.

> This time I made sure I wasn't on gm mode and I was still able to learn goblin engineering recipies even though the char wasn't a goblin engineer <- this is a core thing isn't it?

Both issues still present as of early 2019. Both of these issues are core issues aren't they @cala ? - should we close this and raise new issues there?
